### PR TITLE
update readme with steps for openshift deployment option

### PR DESCRIPTION
**Pull request by kevprice83**
This PR will update the READme as well as add the redis template necessary for the OpenShift deployment.

### CODE REVIEW

File: doc/oauth-guide.md [hunk 1]
```diff
@@ -33,6 +33,32 @@ Here we have added the `-v` flag to run APIcast in verbose mode, in order to all
...
+
+### Steps
+
+The following commands are based on the **oc cluster up** installation.
+
+1. Log in as admin and create a persistent volume using `oc login -u system:admin` and `redis-persistent.json`
+  - mkdir -p /var/lib/docker/pv/01
+  - chmod g+w /var/lib/docker/pv/01
+  - chcon -Rt svirt_sandbox_file_t /var/lib/docker/pv/
+  - oc new-app --param PV=01 -f redis-persistent.json
```
Comment (mikz) [inline doc/oauth-guide.md]: This is not going to work. There are no persistent volumes set up.

Comment (kevprice83) [inline doc/oauth-guide.md]: This is the persistent volume

Comment (mikz) [inline doc/oauth-guide.md]: No. This is template that uses Persistent Volumes by Persistent Volume Claims. No Persistent Volumes are created and have to be created before creating Persistent Volume Claims.

Comment (kevprice83) [inline doc/oauth-guide.md]: Okay I see what you mean, I pushed the wrong version of the template I was using locally. My bad...

File: openshift/redis-persistent.json [hunk 2]
```diff
@@ -65,6 +65,25 @@
...
+      "spec": {
+        "accessModes": [
+          "ReadWriteOnce",
+          "ReadWriteMany"
+        ],
+        "capacity": {
+          "storage": "10Gi"
+        },
+        "persistentVolumeReclaimPolicy": "Recycle",
+        "hostPath": {
```
Comment (mikz) [inline openshift/redis-persistent.json]: This makes the template not really reusable between installations. PersistentVolumes are usually not created in application templates. Using `hostPath` will lock the deployment to one node and is not production ready.

File: doc/oauth-guide.md [hunk 3]
```diff
@@ -33,6 +33,32 @@ Here we have added the `-v` flag to run APIcast in verbose mode, in order to all
...
+- Cluster is up and running
+- New project has been created
+
+Proceed to the following steps if you have fulfilled the 2 previous requirements and then continue the set up as described in our [OpenShift tutorial](https://support.3scale.net/docs/deployment-options/apicast-openshift)
+
+### Steps
+
+The following commands are based on the **oc cluster up** installation.
+
+1. Log in as admin and create a persistent volume using `oc login -u system:admin` and `redis-persistent.json`
```
Comment (mikz) [inline doc/oauth-guide.md]: Are you sure persistent is needed? From what I remember it is just used for few seconds to hold some state and ephemeral would be just fine.

Comment (kevprice83) [inline doc/oauth-guide.md]: I guess so. I tested ephemeral and then implemented persistent based on your advice :) I can certainly re-do this commit to only use the ephemeral Redis as it is only storing keys with state values for the OAuth handshake. I'm really not that up-to-speed with OpenShift and Redis so all advice is appreciated here.

Comment (mikz) [inline doc/oauth-guide.md]: I suggested using ephemeral first and then going for persistence - if you need persistence. 
Do you need persistence? I don't think so. So just use ephemeral.

Comment (kevprice83) [inline doc/oauth-guide.md]: Yeah it seems Redis is used for the OAuth handshake only in this case, anyway, I saw it mentioned somewhere that the Redis dependency will be removed from OAuth altogether, is that correct?

Comment (mikz) [inline doc/oauth-guide.md]: Yes. https://github.com/3scale/apicast/issues/174

**Comment by mpguerra** [comment]
@kevprice83 do you still need my review on this? Or do you need to update based on comments by @mikz 

Would be good to update the support site with these additional details once ready

**Comment by kevprice83** [comment]
Closing due to rewrite of the OAuth integration pending.

**Pull request closed without merging**